tests/mbt/driver/core.go

```diff
@@ -0,0 +1,418 @@
+	simibc "github.com/cosmos/interchain-security/v3/testutil/simibc"
+	consumerkeeper "github.com/cosmos/interchain-security/v3/x/ccv/consumer/keeper"
+	consumertypes "github.com/cosmos/interchain-security/v3/x/ccv/consumer/types"
+	providerkeeper "github.com/cosmos/interchain-security/v3/x/ccv/provider/keeper"
+	providertypes "github.com/cosmos/interchain-security/v3/x/ccv/provider/types"
+)
+
+// Define a new type for ChainIds to be more explicit
+// about what the string represents.
+type ChainId string
+
+type ModelParams struct {
+	VscTimeout              time.Duration
+	CcvTimeout              map[ChainId]time.Duration
+	UnbondingPeriodPerChain map[ChainId]time.Duration
+	TrustingPeriodPerChain  map[ChainId]time.Duration
+	MaxClockDrift           time.Duration
+}
+
+type Driver struct {
+	t *testing.T
+
+	coordinator *ibctesting.Coordinator
+
+	// simulate IBC network: for each consumer chain name, we have a path between consumer and provider
+	simibcs map[ChainId]*simibc.RelayedPath
+
+	// keep around validators for easy access
+	validators []*cmttypes.Validator
+	valNames   []string
+
+	DriverStats *Stats
+}
+
+// ctx returns the sdk.Context for the chain
+func (s *Driver) ctx(chain ChainId) sdk.Context {
+	return s.chain(chain).GetContext()
+}
+
+// returns the path from the given chain to the provider.
+func (s *Driver) path(chain ChainId) *simibc.RelayedPath {
+	return s.simibcs[chain]
+}
+
+// chain returns the TestChain for a given chain identifier
+func (s *Driver) chain(chain ChainId) *ibctesting.TestChain {
+	return s.coordinator.GetChain(string(chain))
+}
+
+func (s *Driver) providerChain() *ibctesting.TestChain {
+	return s.chain("provider")
+}
+
+func (s *Driver) providerCtx() sdk.Context {
+	return s.providerChain().GetContext()
+}
+
+func (s *Driver) providerKeeper() providerkeeper.Keeper {
+	return s.providerChain().App.(*appProvider.App).ProviderKeeper
+}
+
+func (b *Driver) providerStakingKeeper() stakingkeeper.Keeper {
+	return *b.providerChain().App.(*appProvider.App).StakingKeeper
+}
+
+func (b *Driver) consumerKeeper(chain ChainId) consumerkeeper.Keeper {
+	return b.chain(chain).App.(*appConsumer.App).ConsumerKeeper
+}
+
+// runningTime returns the timestamp of the current header of chain
+func (s *Driver) runningTime(chain ChainId) time.Time {
+	testChain := s.chain(chain)
+	return testChain.CurrentHeader.Time
+}
+
+// lastTime returns the timestamp of the last header of chain
+func (s *Driver) lastTime(chain ChainId) time.Time {
+	testChain := s.chain(chain)
+	return testChain.LastHeader.Header.Time
+}
+
+// delegator retrieves the address for the delegator account
+func (s *Driver) delegator() sdk.AccAddress {
+	return s.providerChain().SenderAccount.GetAddress()
+}
+
+// validator returns the address for the validator with id (ix) i
+func (s *Driver) validator(i int64) sdk.ValAddress {
+	return sdk.ValAddress(s.validators[i].Address)
+}
+
+// consumerPower returns the power on the consumer chain chain for
```

Comment:
```suggestion
// consumerPower returns the power on the consumer chain for
```
Would it make sense to just write "index" instead of "id (ix)"?